woocommerce/woocommerce-android
# feat: duplicate products

Closes: #8157


### Description

This PR adds a feature of duplicating products - including variable products.

iOS reference: https://github.com/woocommerce/woocommerce-ios/pull/7727

Firstly review&merge FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2629

### Testing instructions

The feature is self-explanatory, please test it as you were a user. Interesting corner cases:
- duplicate product that has more than 100 variations
- verify that if you customize a variation (add description or change image) and then you duplicate a product, variations in duplicated product are also customized (have the same description or image). In other words, please check if https://github.com/woocommerce/woocommerce-ios/issues/8611 doesn't happen.

### Images/gif


https://user-images.githubusercontent.com/5845095/212022201-dcea928f-b066-4d63-b162-29cf08421927.mp4
